### PR TITLE
  Simplify virtualenv testing

### PR DESCRIPTION
**Pull request by kubouch**
# Description

(Resurrect of https://github.com/nushell/nushell/pull/9521)

Instead of running virtualenv's test suite, run only Nushell's test script (not including pydoc test, not sure how to get it working without the virtualenv's test suite).

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -115,21 +115,12 @@ jobs:
         with:
           python-version: "3.10"
 
-      - run: python -m pip install tox
-
       - name: Install virtualenv
-        run: git clone https://github.com/pypa/virtualenv.git
+        run: run: pip install virtualenv
```
Comment (sholderbach) [inline .github/workflows/ci.yml:119]:
```suggestion
        run: pip install virtualenv
```

**Comment by kubouch** [comment]
Thanks! Somehow I didn't see that...

**Comment by amtoine** [comment]
almost there :star_struck:

**Comment by kubouch** [comment]
`UnicodeEncodeError: 'charmap' codec can't encode characters in position 26-31: character maps to <undefined>` I have no idea how to fix that.

**Comment by amtoine** [comment]
> `UnicodeEncodeError: 'charmap' codec can't encode characters in position 26-31: character maps to <undefined>` I have no idea how to fix that.
>
it's strange because this PR shouldn't change the set of tests being run, right?
and the tests work on `main` :thinking:

**Comment by kubouch** [comment]
Maybe it's some UTF-16 encoding related issue on Windows... Not sure...

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -115,21 +115,12 @@ jobs:
...
 
       - name: Test Nushell in virtualenv
-        run: |
-          cd virtualenv
-          # if we encounter problems with bleeding edge tests pin to the latest tag
-          # git checkout $(git describe --tags | cut -d - -f 1)
-          # We need to disable failing on coverage levels.
-          nu -c "open pyproject.toml | upsert tool.coverage.report.fail_under 1 | save patchproject.toml"
-          mv patchproject.toml pyproject.toml
-          tox -e ${{ matrix.py }} -- -k nushell
```
Comment (amtoine) [inline .github/workflows/ci.yml:132]: i do not see `tox` tests in the new `test_virtualenv.nu`, is that normal?

**Review by amtoine — DISMISSED** [review]
almost there :partying_face: 
well done with the encoding stuff :muscle:

File: scripts/test_virtualenv.nu [hunk 3]
```diff
@@ -0,0 +1,57 @@
+if $nu.os-info.family == 'windows' {
+    # fix encoding on Windows https://stackoverflow.com/a/63573649
+    load-env {
+        PYTHONIOENCODING: utf-8
+        PYTHONLEGACYWINDOWSSTDIO: utf-8
+    }
+}
+
+let env_name = 'e-$ èрт🚒♞中片-j'
```
[the remark this answers is not in the document]
Comment (kubouch) [reply scripts/test_virtualenv.nu:9]: Yes, taken from virtualenv test suite

File: scripts/test_virtualenv.nu [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+let paths = if $nu.os-info.family == 'windows' {
+    ['Scripts', 'python.exe']
+} else {
+    ['bin', 'python']
+}
+
+let subdir = $paths.0
```
[the remark this answers is not in the document]
Comment (kubouch) [reply scripts/test_virtualenv.nu:17]: Exactly

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kubouch) [reply .github/workflows/ci.yml:132]: tox is Python test runner used in virtualenv repo to run the tests. We don't need it.

**Comment by kubouch** [comment]
The encoding thing feels like a hack, but it might be OK for just testing if the activation script succeeds.

File: scripts/test_virtualenv.nu (hunk 3 above)
[the remark this answers is not in the document]
Comment (amtoine) [reply scripts/test_virtualenv.nu:9]: ok :+1:

File: .github/workflows/ci.yml (hunk 2 above)
Comment (amtoine) [reply .github/workflows/ci.yml:132]: ooh okok :+1:

**Review by sholderbach — APPROVED** [review]
Thanks for reworking this!
The script is more readable than the whole tox -> coverage -> pytest system of the original test suite.

For me a remaining question is how we handle breaking changes now, when we update their activation script in anticipation of our breaking changes but it is not published on PyPI yet.
It keeps us honest about language breaking changes but could be a bit restrictive in some cases.

**Comment by kubouch** [comment]
Currently, the tests simulate the common user experience with `pip install virtualenv`. Usually, virtualenv releases are quite fast, so it might be OK for us to wait for them. If it becomes a problem, we can clone the virtualenv main branch.

**Comment by kubouch** [comment]
OK, let's test it and see how it goes.

**Pull request merged**